components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -466,22 +591,47 @@ private List<PerformableOperation> validatePerformableOperations(ActionExecution
                                 performableOperation)))
                 .collect(Collectors.toList());
 
-        if (LOG.isDebugEnabled()) {
             // todo: add to diagnostics
-            List<String> allowedOps = new ArrayList<>();
-            List<String> notAllowedOps = new ArrayList<>();
-
-            response.getOperations().forEach(operation -> {
-                String operationDetails = "Operation: " + operation.getOp() + " with path: " + operation.getPath();
-                if (allowedPerformableOperations.contains(operation)) {
-                    allowedOps.add(operationDetails);
-                } else {
-                    notAllowedOps.add(operationDetails);
+            if (LOG.isDebugEnabled() || LoggerUtils.isDiagnosticLogsEnabled()) {
+                List<String> allowedOps = new ArrayList<>();
+                List<String> notAllowedOps = new ArrayList<>();
+
+                response.getOperations().forEach(operation -> {
+                    String operationDetails = "Operation: " + operation.getOp() + " with path: " + operation.getPath();
+                    if (allowedPerformableOperations.contains(operation)) {
+                        allowedOps.add(operationDetails);
+                    } else {
+                        notAllowedOps.add(operationDetails);
+                    }
+                });
+                LOG.debug("Allowed Operations: " + String.join(", ", allowedOps) +
+                        ". Not Allowed Operations: " + String.join(", ", notAllowedOps));
```

Comment:
This is not needed right ? There's a debug log added below if debug is enabled

Reply:
Updated in